src/psyclone/domain/lfric/lfric_extract_driver_creator.py

```diff
@@ -915,12 +865,40 @@ def create(self, nodes, input_list, output_list,
 
         return file_container
 
+    # -------------------------------------------------------------------------
+    @staticmethod
+    def collect_all_required_modules(file_container):
+        '''Collects recursively all modules used in the file container.
+        It returns a dictionary, with the keys being all the (directly or
+        indirectly used modules)
+
+        :param file_container:
+        :type file_container: \
+            :py:class:`psyclone.psyir.psyir.nodes.FileContainer`
+
+        :returns: a dictionary, with the required module names as key, and \
+            as value a set of all modules required by the key module.
+        :rtype: Set[str]
+
+        '''
+        sym_tab = file_container.children[0].symbol_table
+
+        # Add all modules imported, except intrinsic ones
+        all_mods = set()
+        for symbol in sym_tab.symbols:
+            if isinstance(symbol, ContainerSymbol) and \
+                    ",intrinsic" not in symbol.name:
```

Comment:
This doesn't look right but it's probably not your fault. How come the symbol name has ",intrinisc" in it?

Reply:
OK, I could not find an example where this is actually required anymore - can't say if I maybe had old files around (since before this PR a driver would contain an import  from iso_fortran_env. I tried various gungho files, and none of them triggered this (since this is basically running on the PSyclone created PSy layer only anyway). So I have just removed the second test.